File: Entregables/Palao Rocio/copy_mov/copy_mov_java/Java_Copy_Mov/src/main/java/org/example/controller/FicherosController.java
package org.example.controller;

import org.example.utils.Input;

import java.io.File;
import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;
import java.nio.file.Paths;
import java.nio.file.StandardCopyOption;

public class FicherosController {
    private final Input input;


    /**
     * Constructor
     * @param input Clase necesaria para pedir datos
     */
    public FicherosController(Input input) {
        this.input = input;
    }


    /**
     * Para preguntar por el comando, y realizar todas las acciones.
     */
    public void startCommand(){
        var line =input.getStringPath();
        var separado = line.split(" ");
        Path inicialPath = Paths.get(separado[1]).toAbsolutePath();
        Path finalPath = Paths.get(separado[2]).toAbsolutePath();

        if(rutaInicialCorrecta(inicialPath)) {
            selectComand(separado[0], inicialPath, finalPath);
        }
    }


    /**
     * Para saber si la ruta inicial dada existe.
     * @param inicialPath ruta inicial.
     * @return si existe o no.
     */
    private boolean rutaInicialCorrecta(Path inicialPath) {
        return Files.exists(inicialPath);
    }


    /**
     * Crear la ruta final.
     * @param line ruta final.
     */
    private void createFinalPath(Path line) {
       if (Files.notExists(line)){
           try {
               Files.createDirectories(line);
           } catch (IOException e) {
               throw new RuntimeException(e);
           }
       }
    }


    /**
     * Para seleccionar si es copiar o mover.
     * @param comando contiene si es copiar o mover.
     * @param inicialPath ruta inicial del fichero.
     * @param finalPath ruta final del fichero.
     */
    private void selectComand(String comando, Path inicialPath, Path finalPath){
        if (comando.equals("cp")){
            makeCp(inicialPath, finalPath);
        }else{
            makeMv(inicialPath, finalPath);
        }
    }


    /**
     * Mover el fichero de la ruta inicial a la ruta final.
     * @param rutaInicial ruta inicial del fichero.
     * @param rutaFinal ruta final del fichero.
     */
    private void makeMv(Path rutaInicial, Path rutaFinal) {
        try {
            Files.move(rutaInicial, rutaFinal, StandardCopyOption.REPLACE_EXISTING);
        } catch (IOException e) {
            throw new RuntimeException(e);
        }
    }


    /**
     * Copiar el fichero desde la ruta inicial hasta la ruta final.
     * @param rutaInicial ruta inicial del fichero.
     * @param rutaFinal ruta final del fichero.
     */
    private void makeCp(Path rutaInicial, Path rutaFinal) {
        if(Files.exists(rutaFinal)){
            if (!input.sobreescribirArchivo()){
                System.out.println("No sé copió el archivo");
                return;
            }
        }else{
            createFinalPath(rutaFinal);
        }

        try {
            Files.copy(rutaInicial,rutaFinal,StandardCopyOption.REPLACE_EXISTING);
        } catch (IOException e) {
            throw new RuntimeException(e);
        }
    }

}

File: Entregables/Palao Rocio/copy_mov/cp_mv_kotlin/src/main/kotlin/controller/FicherosController.kt
package controller

import utils.Input
import java.nio.file.Files
import java.nio.file.Path
import java.nio.file.Paths
import java.nio.file.StandardCopyOption

class FicherosController(private val inputs: Input) {

    /**
     * Para preguntar por el comando, y realizar todas las acciones.
     */
    init{
        val line = inputs.getStringPath()
        val separated = line.split(" ")
        val inicialPath = Paths.get(separated[1]).toAbsolutePath()
        val finalPath = Paths.get(separated[2]).toAbsolutePath()

        if (inicialPath.rutaInicialCorrecta()){
            selectedCommand(separated[0], inicialPath, finalPath)
        }
    }


    /**
     * Para seleccionar si es copiar o mover.
     * @param command contiene si es copiar o mover.
     * @param inicialPath ruta inicial del fichero.
     * @param finalPath ruta final del fichero.
     */
    private fun selectedCommand(command: String, inicialPath: Path, finalPath: Path) {
        when(command){
            "cp" -> makeCp(inicialPath, finalPath)
            "mv" -> makeMv(inicialPath,finalPath)
        }
    }


    /**
     * Mover el fichero de la ruta inicial a la ruta final.
     * @param inicialPath ruta inicial del fichero.
     * @param finalPath ruta final del fichero.
     */
    private fun makeMv(inicialPath: Path, finalPath: Path) {
        if (!Files.exists(finalPath)) {
            finalPath.createFinalPath()
        }
        Files.move(inicialPath, finalPath, StandardCopyOption.REPLACE_EXISTING)
    }


    /**
     * Copiar el fichero desde la ruta inicial hasta la ruta final.
     * @param inicialPath ruta inicial del fichero.
     * @param finalPath ruta final del fichero.
     */
    private fun makeCp(inicialPath: Path, finalPath: Path) {
        if (Files.exists(finalPath)){
            if (!inputs.sobreescribirArchivo()){
                println("No se copió el archivo")
                return
            }
        }else{
            finalPath.createFinalPath()
        }

        Files.copy(inicialPath, finalPath, StandardCopyOption.REPLACE_EXISTING)
    }




    /**
     * Función de extensión.
     * Crear la ruta final.
     */
    private fun Path.createFinalPath() {
        if (Files.notExists(this)){
            Files.createDirectories(this)
        }
    }


    /**
     * Función de extensión.
     * Para saber si la ruta inicial dada existe.
     * @return si existe o no.
     */
    private fun Path.rutaInicialCorrecta(): Boolean {
        return Files.exists(this)
    }
}